Move implementation of CU err handling to Exceptions.cpp

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72306
* __->__ #72958

Please note, that it does not depend on any of the symbols from CUDA libraries

Differential Revision: [D34296428](https://our.internmc.facebook.com/intern/diff/D34296428)

malfet committed:
Move implementation of CU err handling to Exceptions.cpp

Please note, that it does not depend on any of the symbols from CUDA libraries

[ghstack-poisoned]

zasdfgbnm commented on aten/src/ATen/cuda/Exceptions.cpp:
```diff
@@ -0,0 +1,68 @@
+//NS: CUDACachingAllocator must be included before to get CUDART_VERSION definedi
+#include <c10/cuda/CUDACachingAllocator.h>
```
Should we include `cuda_runtime_api.h` instead?

malfet replied:
I'm not sure will it be correctly hipified

malfet committed:
Update on "Move implementation of CU err handling to Exceptions.cpp"


Please note, that it does not depend on any of the symbols from CUDA libraries

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34296428).
